tests/wp-unit/include/Core/Posts/Modules/DeletePostsByURLModuleTest.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace BulkWP\BulkDelete\Core\Posts\Modules;
+
+use BulkWP\Tests\WPCore\WPCoreUnitTestCase;
+
+/**
+ * Test Deletion of Posts by URL.
+ *
+ * Tests \BulkWP\BulkDelete\Core\Posts\Modules\DeletePostsByURLModule
+ *
+ * @since 6.0.0
+ */
+class DeletePostsByURLModuleTest extends WPCoreUnitTestCase {
+
+	/**
+	 * The module that is getting tested.
+	 *
+	 * @var \BulkWP\BulkDelete\Core\Posts\Modules\DeletePostsByURLModule
+	 */
+	protected $module;
+
+	public function setUp() {
+		parent::setUp();
+
+		$this->module = new DeletePostsByURLModule();
+	}
+
+	/**
+	 * Helper function.
+	 * create post by URL
+	 */
+	public function get_post_by_url( $url ) {
```

Comment:
Add more details in PHPDoc for this helper function. We will move it into the test library.